packages/binding-mqtt/src/mqtt-broker-server.ts

```diff
@@ -300,7 +306,11 @@ export default class MqttBrokerServer implements ProtocolServer {
             };
 
             try {
-                thing.handleWriteProperty(segments[3], JSON.parse(payload.toString()), options);
+                thing.handleWriteProperty(
+                    segments[3],
+                    new Content(ContentSerdes.DEFAULT, Readable.from(payload.toString())),
```

Comment:
Question: Should it be 
new Content(ContentSerdes.DEFAULT, Readable.from(JSON.parse(payload.toString()))
as it used to be in 
https://github.com/eclipse/thingweb.node-wot/blob/09dfd6ab03805b7b39ab39433880496466d25e67/packages/binding-mqtt/src/mqtt-broker-server.ts#L303

Please check.

Reply:
Note: this might need checks/tests especially with a variation of input: simple types (int, boolean, etc) vs composed types (like objects and arrays)

Reply:
Checking now

Reply:
Passing JSON object to `Readable.from` creates an issue except when JSON only a string, therefore it seems like we should not use `JSON.parse`.

Reply:
Thanks for checking 👍
We should add tests in follow-up PR for it.